src/Controllers/HelloWorldController.cs

```diff
@@ -1,26 +1,31 @@
 using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Localization;
 using System.Text.Encodings.Web;
 
 namespace TrainingNet.Controllers
 {
     [Route("[controller]")]
     public class HelloWorldController : Controller
     {
-        // GET: /HelloWorld/
+        private readonly IHtmlLocalizer<HomeController> _localizer;
+
+        public HelloWorldController(IHtmlLocalizer<HomeController> localizer)
+        {
+            this._localizer = localizer;
+        }
+
         [HttpGet("")]
         [HttpGet("Index")]
         public IActionResult Index()
         {
             return View();
         }
 
-        // GET: /HelloWorld/Welcome/ 
         [HttpGet("Welcome/{id?}")]
         public IActionResult Welcome(string name, int numTimes = 1)
         {
-            ViewData["Message"] = "Hello " + name;
+            ViewData["Message"] = _localizer["Greeting"].Value + name;
```

Comment:
Try using a property instead of using the localizer directly.

`public IHtmlLocalizer Localizer {
    get => return this._localizer;
}`